app/src/main/java/de/westnordost/streetcomplete/tangram/QuestsMapFragment.java

```diff
@@ -385,6 +385,25 @@ public void addQuest(Quest quest, QuestGroup group)
 	}
 */
 
+	protected Long questPriority(Quest quest){
+		// priority is decided by
+		// - primarily by quest type to allow quest prioritization
+		// - for quests of the same type - influenced by quest id,
+		//   this is done to reduce chance that as user zoom in a quest disappears,
+		//   especially in case where disappearing quest is one that user selected to solve
+
+		// main priority part - values fit into Integer, but with as large steps as possible
+		Integer order = questTypeOrder.get(quest.getType());
+		if(order == null) order = 0;
+		Integer freeValuesForEachQuest = Integer.MAX_VALUE / questTypeOrder.size();
```

Comment:
```suggestion
		int freeValuesForEachQuest = Integer.MAX_VALUE / questTypeOrder.size();
```

Usually the classes Integer, Long etc are only used when the value can be null.